Automattic/simplenote-android
# Add Dark Widget

### Fix
Add a dark version of the note widget.  See the screenshots below for illustration.

<a href="https://user-images.githubusercontent.com/3827611/74388761-7a0a3800-4db9-11ea-8c61-f7c11eceee2e.png"><img src="https://user-images.githubusercontent.com/3827611/74388761-7a0a3800-4db9-11ea-8c61-f7c11eceee2e.png" width="300"></a>

#### Note
These changes have a lot of duplicated code since this was a small, time-boxed project.  Ideally, the duplication would be consolidated, but we want to see more usage of the note widgets before investing time into optimizing the code.

### Test
These steps assume the AOSP launcher (i.e. Nexus/Pixel launcher). The steps to add the widget to the home screen may be slightly different based on the launcher used. Other steps should be the same regardless of launcher.
1. Long-press home screen.
2. Tap ***Widgets*** option.
3. Scroll to ***Simplenote*** widget.
4. Notice ***Note (Dark)*** and ***Note (Light)*** widgets with ***To-Do List*** preview images.
5. Long-press ***Note (Dark)*** widget.
6. Place ***Note (Dark)*** widget on home screen.
7. Notice ***Select Note*** dialog with note list.
8. Tap note in list.
9. Notice note title shown in widget with dark theme.
10. Long-press widget.
11. Notice circular handles are shown.
12. Drag circular handles horizontally and vertically.
13. Notice note title and content are shown in widget with dark theme when large enough.
14. Tap widget.
15. Notice note is shown in app.
16. Go to ***All Notes***.
17. Delete note from Step 8.
18. Go to ***Trash***.
19. Empty trash.
20. Go to home screen.
21. Notice ***Note not found*** shown in widget.
22. Tap widget.
23. Notice ***Select Note*** dialog with note list.
24. Tap ***Cancel*** button to dismiss dialog.
25. Go to ***Settings***.
26. Tap ***Log out*** option.
27. Notice login screen shown.
28. Go to home screen.
29. Notice ***Log in to use widget*** shown in widget.
30. Tap widget.
31. Notice login screen shown.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in 4da9b94a with:
> Added a dark version of the note widget to view and open a note in the app from the home screen
